src/core/packages/elasticsearch/server-internal/src/elasticsearch_service.test.ts

```diff
@@ -224,243 +227,274 @@ describe('#setup', () => {
 
   it('esNodeVersionCompatibility$ only starts polling when subscribed to', async () => {
     const mockedClient = mockClusterClientInstance.asInternalUser;
-    mockedClient.nodes.info.mockImplementation(() =>
-      elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
-    );
+    // mocking successful response becuase error responses would trigger retries and complicate the test
+    mockedClient.nodes.info.mockResolvedValue({
+      nodes: {},
+    } as any);
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(0);
 
     const setupContract = await elasticsearchService.setup(setupDeps);
 
+    // Polling starts immediately when setup subscribes
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
 
     tick();
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
 
     await firstValueFrom(setupContract.esNodesCompatibility$);
-    expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2); // shares the last value
+    // Should reuse the cached value from the first subscription
+    expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
   });
 
   it('esNodeVersionCompatibility$ stops polling when unsubscribed from', async () => {
     const mockedClient = mockClusterClientInstance.asInternalUser;
-    mockedClient.nodes.info.mockImplementation(() =>
-      elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
-    );
+    mockedClient.nodes.info.mockResolvedValue({
```

Comment:
This test only checks that polling stops on unsubscription, we don't need the complexity of retrying requests for that.